p2p/src/net/mock/transport/tcp/adapter/noise.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use snowstorm::NoiseStream;
+use tokio::net::TcpStream;
+
+use crate::{error::P2pError, net::mock::peer::Role};
+
+use super::StreamAdapter;
+
+static NOISE_HANDSHAKE_PATTERN: &str = "Noise_XX_25519_ChaChaPoly_SHA256";
+
+static NOISE_HANDSHAKE_PARAMS: once_cell::sync::Lazy<snowstorm::NoiseParams> =
+    once_cell::sync::Lazy::new(|| NOISE_HANDSHAKE_PATTERN.parse().expect("valid pattern"));
+
+#[derive(Debug)]
+pub struct NoiseEncryptionAdapter {}
```

Comment:
Why not put the key in here, and ditch the whole new `StreamConfig` object?

Reply:
Because `StreamAdapter` is only relevant to `TcpMockTransport`.
It is irrelevant to `ChannelMockTransport` because `ChannelMockTransport` does not work with `TcpStream`s.
But we still need somehow pass own keys from `MockTransport` to `TcpMockTransport`.
That's why I added generic `StreamConfig` type to `MockTransport`.

Reply:
I tried again but don't see why it would be any better.
We would need to pass `StreamAdapter` around (instead of `StreamKey`) and it simply adds more code.